src/OverlayTrigger.js

```diff
@@ -98,97 +70,73 @@ const defaultProps = {
   trigger: ['hover', 'focus'],
 };
 
+// eslint-disable-next-line react/no-multi-comp
 class OverlayTrigger extends React.Component {
   constructor(props, context) {
     super(props, context);
 
-    this.handleToggle = this.handleToggle.bind(this);
-    this.handleDelayedShow = this.handleDelayedShow.bind(this);
-    this.handleDelayedHide = this.handleDelayedHide.bind(this);
-    this.handleHide = this.handleHide.bind(this);
-
-    this.handleMouseOver = e =>
-      this.handleMouseOverOut(this.handleDelayedShow, e, 'fromElement');
-    this.handleMouseOut = e =>
-      this.handleMouseOverOut(this.handleDelayedHide, e, 'toElement');
-
-    this._mountNode = null;
-
+    this.trigger = React.createRef();
     this.state = {
-      show: props.defaultOverlayShown,
+      show: props.defaultShow,
     };
   }
 
-  componentDidMount() {
-    this._mountNode = document.createElement('div');
-    this.renderOverlay();
+  componentWillUnmount() {
+    clearTimeout(this._timeout);
   }
 
-  componentDidUpdate() {
-    this.renderOverlay();
+  getChildProps() {
+    return React.Children.only(this.props.children).props;
   }
 
-  componentWillUnmount() {
-    ReactDOM.unmountComponentAtNode(this._mountNode);
-    this._mountNode = null;
+  getTarget = () => this.trigger.current;
 
-    clearTimeout(this._hoverShowDelay);
-    clearTimeout(this._hoverHideDelay);
+  hide() {
+    this.setState({ show: false });
   }
 
-  handleDelayedHide() {
-    if (this._hoverShowDelay != null) {
-      clearTimeout(this._hoverShowDelay);
-      this._hoverShowDelay = null;
-      return;
-    }
+  show() {
+    this.setState({ show: true });
+  }
 
-    if (!this.state.show || this._hoverHideDelay != null) {
-      return;
-    }
+  handleShow = () => {
```

Comment:
this doesn't quite match the old logic...

in the old logic, if there already was a show timeout pending, then we wouldn't reset the timeout. we'd just... do nothing.

this makes it so that if a "handleShow" triggers while we have a pending show, we reset the timer.

suppose the triggers were "hover" and "click". if you clicked after hovering, the new logic would reset the timeout, which i do not think is what we want

Reply:
That's true, I didn't notice the logic change. Hmm I think I mostly copied this from what upstream does, so I agree our old behavior is seems more correct but it'd be divergent, unless I copied it wrong